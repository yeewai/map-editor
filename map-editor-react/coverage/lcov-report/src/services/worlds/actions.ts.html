<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/services/worlds/actions.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/services/worlds</a> actions.ts
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>24/24</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>15/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>23/23</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">24x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">20x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">18x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">19x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { ActionCreator, Dispatch } from 'redux';
import { ThunkAction } from 'redux-thunk';
import { World } from './types';
import { StateTree, Action } from 'services/types';
import * as api from 'api';
import _ from 'lodash';
&nbsp;
import { modalActions } from '@evercourse/ever-modal';
import { structureDefinitionsSelectors } from 'services/structureDefinitions';
&nbsp;
export const requestWorlds: ActionCreator&lt;Action&gt; = () =&gt; ({
    type: 'world/REQUEST'
});
&nbsp;
export const receiveWorlds: ActionCreator&lt;Action&gt; = ( worlds: World[] )=&gt; ({
    type: 'world/RECEIVE',
    payload: worlds
});
&nbsp;
export const setError: ActionCreator&lt;Action&gt; = (jqXHR: any, textStatus: string, errorThrown: any) =&gt; ({
    type: 'world/SET_ERROR',
    payload: { jqXHR, textStatus, errorThrown }
});
&nbsp;
//I have no idea what the last return type is so I set it to any.........
export const fetchWorlds: ActionCreator&lt;ThunkAction&lt;any, StateTree, void&gt;&gt; = () =&gt; {
    return (dispatch: Dispatch&lt;StateTree&gt;): any =&gt; {
        dispatch(requestWorlds());
        return api.getWorlds().then (
            resp =&gt; dispatch(receiveWorlds(resp)),
            (jqXHR, textStatus, errorThrown) =&gt;{
                 dispatch(setError(jqXHR, textStatus, errorThrown))
             }
        );
    };
};
&nbsp;
const putWorld: ActionCreator&lt;Action&gt; = () =&gt; ({
    type: 'world/PUT'
});
&nbsp;
export const updateWorld: ActionCreator&lt;ThunkAction&lt;any, StateTree, void&gt;&gt; = (values) =&gt; {
    return (dispatch: Dispatch&lt;StateTree&gt;, getState: () =&gt; StateTree ) =&gt; {
&nbsp;
        dispatch(putWorld());
        return dispatch(processWorld(values));
    };
}
&nbsp;
const createWorld: ActionCreator&lt;Action&gt; = () =&gt; ({
    type: 'world/CREATE'
});
&nbsp;
export const addWorld: ActionCreator&lt;ThunkAction&lt;any, StateTree, void&gt;&gt; = (values) =&gt; {
    return (dispatch: Dispatch&lt;StateTree&gt;, getState: () =&gt; StateTree ) =&gt; {
&nbsp;
        dispatch(createWorld());
&nbsp;
        return dispatch(processWorld(values));
    };
}
&nbsp;
// const processingWorld: ActionCreator&lt;Action&gt; = () =&gt; ({
//     type: 'world/PROCESS'
// });
export const processWorld: ActionCreator&lt;ThunkAction&lt;any, StateTree, void&gt;&gt; = (values) =&gt; {
    return (dispatch: Dispatch&lt;StateTree&gt;, getState: () =&gt; StateTree ) =&gt; {
&nbsp;
        return api.createWorld(values).then(
            resp =&gt; {
                dispatch(modalActions.hideModal());
                dispatch(fetchWorlds());
            },
            (jqXHR, textStatus, errorThrown) =&gt; dispatch(setError(jqXHR, textStatus, errorThrown))
        );
    };
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 21 2018 10:53:31 GMT-0600 (CST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
