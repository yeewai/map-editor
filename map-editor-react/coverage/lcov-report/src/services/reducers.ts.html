<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/services/reducers.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/services</a> reducers.ts
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/5</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { rout<span class="cstat-no" title="statement not covered" >erReducer } from 'react-route<span class="cstat-no" title="statement not covered" >r</span>-redux';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ></span></span></span>
import { comb<span class="cstat-no" title="statement not covered" >ineReducers } fro</span>m 'redux';
<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span>
import { StateTree<span class="cstat-no" title="statement not covered" > } from './types';</span>
import structureDefinitions from './structureDefinitions';
<span class="cstat-no" title="statement not covered" ></span>
const rootReducer = combineReducers&lt;StateTree&gt;({
    router: routerReducer,
    structureDefinitions
});
&nbsp;
export default rootReducer;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jan 31 2018 09:59:49 GMT-0600 (CST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
