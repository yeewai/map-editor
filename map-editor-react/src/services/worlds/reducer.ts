import { Action } from 'services/types';
import * as Types from './types';

export const defaultState: Types.State = {
    isFetching: false,
    hasFetched: false,
    error: undefined,
    items: []
}

const worldReducer = (state = defaultState, action: Action ): Types.State => {
    switch(action.type) {
        case 'world/REQUEST':
            return { ...state, isFetching: true, error: undefined };
        case 'world/RECEIVE':
            return {
                ...state,
                isFetching: false,
                hasFetched: true,
                error: undefined,
                items: action.payload,
            };
        case 'world/SET_ERROR':
            return { ...state, error: 'Error loading world.' };
        default:
            return state;
    }
}


export default worldReducer;
