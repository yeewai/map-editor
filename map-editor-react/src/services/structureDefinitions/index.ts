import reducer from "./reducer";
import * as structureDefinitionsActions from "./actions";
import * as structureDefinitionsSelectors from "./selectors";
import * as structureDefinitionTypes from "./types";

export { structureDefinitionsActions, structureDefinitionsSelectors, structureDefinitionTypes };
export default reducer;
