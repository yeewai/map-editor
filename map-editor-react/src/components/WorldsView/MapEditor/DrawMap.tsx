import * as React from 'react';
import { connect } from "react-redux";

import { TILE_SIZE, INITIAL_ZOOM } from 'conf';
import { OpenModalButton } from '@evercourse/ever-modal';

import DrawBoard from './DrawBoard';
import DrawTile from './DrawTile';
import DrawAddStructureDefinitionTile from './DrawAddStructureDefinitionTile';

import { StateTree } from 'services/types';
import { worldTypes } from 'services/worlds';
import { structureDefinitionTypes } from 'services/structureDefinitions';
import { mapEditorSelectors } from 'services/mapEditor';

interface OwnProps {
    world: worldTypes.WorldWithBoard
}

interface StateProps {
    zoom: number,
    pan: {
        v: number,
        h: number
    }
}

export const mapStateToProps = (state: StateTree, ownProps: OwnProps): StateProps => ({
    zoom: mapEditorSelectors.getZoom(state),
    pan: mapEditorSelectors.getPan(state)
});

export type PropTypes = OwnProps & StateProps;

export const DrawMap: React.SFC<PropTypes> = ( { world, zoom, pan } ) => {
    const viewBox = [
        (pan.h * TILE_SIZE) + (-TILE_SIZE * 30/4),
        (pan.v * TILE_SIZE) + ( TILE_SIZE * 30/2 ) ,
        (TILE_SIZE*INITIAL_ZOOM)/zoom,
        TILE_SIZE*INITIAL_ZOOM/zoom
    ].join(" ");

    return (
        <section>

            <svg className="map" viewBox= {viewBox} preserveAspectRatio="xMinYMax">

                <image xlinkHref={world.bgImageUrl} width={TILE_SIZE*world.width} x={-TILE_SIZE*world.width/2} className="mapBG"/>

                <defs className="structureReference">
                { world.uniqueStructures.map( definition => (
                    <symbol id={definition.id} key={definition.id}
                        viewBox={`0 0 ${definition.imageWidth} ${definition.imageHeight}`}
                        preserveAspectRatio="xMidYMid"
                        dangerouslySetInnerHTML={{__html: definition.image.innerHTML}}
                    />
                )) }
                </defs>


                <DrawBoard board={world.board} />
                <DrawAddStructureDefinitionTile uniqueStructures={ world.uniqueStructures } />

            </svg>
        </section>
    )
};

export default connect<StateProps, any, any>(mapStateToProps)(DrawMap);

// -------- Helpful resources ---------------------------------------------
//http://jsiso.com/tutorials/2014/10/26/isometric-engine-basics.html
//http://clintbellanger.net/articles/isometric_math/
