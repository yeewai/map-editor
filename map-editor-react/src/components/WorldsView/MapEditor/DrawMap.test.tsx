import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import sinon from 'sinon';

import { getMockRouterProps } from 'helpers/test/react-router';
import { DrawMap, mapStateToProps } from './DrawMap';
import { worldTypes } from 'services/worlds';

Enzyme.configure({ adapter: new Adapter() });

describe("Drawing the Map", () => {
    let sandbox, wrapper, routerProps;

    beforeEach(() => {
        sandbox = sinon.createSandbox();
        routerProps = getMockRouterProps<any>(null);
    });

    afterEach(() => {
        sandbox.restore();
    });

    it ("renders", () => {
        const world = {
            uniqueStructures: [
                {
                    ...worldTypes.emptyStructure,
                    id: "a",
                    imageWidth: 1,
                    imageHeight: 2,
                    image: {
                        innerHTML: "Sup"
                    }
                }
            ]
        }

        wrapper = shallow(<DrawMap world={world} zoom={1} pan={ {h: 1, v: 1} } />);
        expect(wrapper).toMatchSnapshot();
    });


    it ("maps state to props", () => {
        const props = mapStateToProps({mapEditor: { zoom: 1, panH: 2, panV: 3}} )

        expect(props.zoom).toEqual(1);
        expect(props.pan).toEqual({h: 2, v: 3});
    })

});
