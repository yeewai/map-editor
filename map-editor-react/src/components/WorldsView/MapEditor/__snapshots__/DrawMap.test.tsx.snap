// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawing the Map renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    pan={
        Object {
            "h": 1,
            "v": 1,
          }
    }
    world={
        Object {
            "uniqueStructures": Array [
              Object {
                "definition": Object {
                  "id": "",
                  "imageUrl": "",
                  "kind": "",
                  "length": 1,
                  "name": "",
                  "width": 1,
                },
                "definitionId": "",
                "id": "a",
                "image": Object {
                  "innerHTML": "Sup",
                },
                "imageHeight": 2,
                "imageWidth": 1,
                "name": "",
                "xposition": 1,
                "yposition": 1,
              },
            ],
          }
    }
    zoom={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <svg
        className="map"
        preserveAspectRatio="xMinYMax"
        viewBox="-65 160 100 100"
>
        <image
                className="mapBG"
                width={NaN}
                x={NaN}
                xlinkHref={undefined}
        />
        <defs
                className="structureReference"
        >
                <symbol
                        dangerouslySetInnerHTML={
                                Object {
                                        "__html": "Sup",
                                      }
                        }
                        id="a"
                        preserveAspectRatio="xMidYMid"
                        viewBox="0 0 1 2"
                />
        </defs>
        <Connect(DrawBoard)
                board={undefined}
        />
        <Connect(Component)
                uniqueStructures={
                        Array [
                                Object {
                                  "definition": Object {
                                    "id": "",
                                    "imageUrl": "",
                                    "kind": "",
                                    "length": 1,
                                    "name": "",
                                    "width": 1,
                                  },
                                  "definitionId": "",
                                  "id": "a",
                                  "image": Object {
                                    "innerHTML": "Sup",
                                  },
                                  "imageHeight": 2,
                                  "imageWidth": 1,
                                  "name": "",
                                  "xposition": 1,
                                  "yposition": 1,
                                },
                              ]
                }
        />
</svg>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <image
            className="mapBG"
            width={NaN}
            x={NaN}
            xlinkHref={undefined}
/>,
          <defs
            className="structureReference"
>
            <symbol
                        dangerouslySetInnerHTML={
                                    Object {
                                                "__html": "Sup",
                                              }
                        }
                        id="a"
                        preserveAspectRatio="xMidYMid"
                        viewBox="0 0 1 2"
            />
</defs>,
          <Connect(DrawBoard)
            board={undefined}
/>,
          <Connect(Component)
            uniqueStructures={
                        Array [
                                    Object {
                                      "definition": Object {
                                        "id": "",
                                        "imageUrl": "",
                                        "kind": "",
                                        "length": 1,
                                        "name": "",
                                        "width": 1,
                                      },
                                      "definitionId": "",
                                      "id": "a",
                                      "image": Object {
                                        "innerHTML": "Sup",
                                      },
                                      "imageHeight": 2,
                                      "imageWidth": 1,
                                      "name": "",
                                      "xposition": 1,
                                      "yposition": 1,
                                    },
                                  ]
            }
/>,
        ],
        "className": "map",
        "preserveAspectRatio": "xMinYMax",
        "viewBox": "-65 160 100 100",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "mapBG",
            "width": NaN,
            "x": NaN,
            "xlinkHref": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "image",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <symbol
                dangerouslySetInnerHTML={
                                Object {
                                                "__html": "Sup",
                                              }
                }
                id="a"
                preserveAspectRatio="xMidYMid"
                viewBox="0 0 1 2"
/>,
            ],
            "className": "structureReference",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "a",
              "nodeType": "host",
              "props": Object {
                "dangerouslySetInnerHTML": Object {
                  "__html": "Sup",
                },
                "id": "a",
                "preserveAspectRatio": "xMidYMid",
                "viewBox": "0 0 1 2",
              },
              "ref": null,
              "rendered": null,
              "type": "symbol",
            },
          ],
          "type": "defs",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "board": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "uniqueStructures": Array [
              Object {
                "definition": Object {
                  "id": "",
                  "imageUrl": "",
                  "kind": "",
                  "length": 1,
                  "name": "",
                  "width": 1,
                },
                "definitionId": "",
                "id": "a",
                "image": Object {
                  "innerHTML": "Sup",
                },
                "imageHeight": 2,
                "imageWidth": 1,
                "name": "",
                "xposition": 1,
                "yposition": 1,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "svg",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <svg
          className="map"
          preserveAspectRatio="xMinYMax"
          viewBox="-65 160 100 100"
>
          <image
                    className="mapBG"
                    width={NaN}
                    x={NaN}
                    xlinkHref={undefined}
          />
          <defs
                    className="structureReference"
          >
                    <symbol
                              dangerouslySetInnerHTML={
                                        Object {
                                                  "__html": "Sup",
                                                }
                              }
                              id="a"
                              preserveAspectRatio="xMidYMid"
                              viewBox="0 0 1 2"
                    />
          </defs>
          <Connect(DrawBoard)
                    board={undefined}
          />
          <Connect(Component)
                    uniqueStructures={
                              Array [
                                        Object {
                                          "definition": Object {
                                            "id": "",
                                            "imageUrl": "",
                                            "kind": "",
                                            "length": 1,
                                            "name": "",
                                            "width": 1,
                                          },
                                          "definitionId": "",
                                          "id": "a",
                                          "image": Object {
                                            "innerHTML": "Sup",
                                          },
                                          "imageHeight": 2,
                                          "imageWidth": 1,
                                          "name": "",
                                          "xposition": 1,
                                          "yposition": 1,
                                        },
                                      ]
                    }
          />
</svg>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <image
              className="mapBG"
              width={NaN}
              x={NaN}
              xlinkHref={undefined}
/>,
            <defs
              className="structureReference"
>
              <symbol
                            dangerouslySetInnerHTML={
                                          Object {
                                                        "__html": "Sup",
                                                      }
                            }
                            id="a"
                            preserveAspectRatio="xMidYMid"
                            viewBox="0 0 1 2"
              />
</defs>,
            <Connect(DrawBoard)
              board={undefined}
/>,
            <Connect(Component)
              uniqueStructures={
                            Array [
                                          Object {
                                            "definition": Object {
                                              "id": "",
                                              "imageUrl": "",
                                              "kind": "",
                                              "length": 1,
                                              "name": "",
                                              "width": 1,
                                            },
                                            "definitionId": "",
                                            "id": "a",
                                            "image": Object {
                                              "innerHTML": "Sup",
                                            },
                                            "imageHeight": 2,
                                            "imageWidth": 1,
                                            "name": "",
                                            "xposition": 1,
                                            "yposition": 1,
                                          },
                                        ]
              }
/>,
          ],
          "className": "map",
          "preserveAspectRatio": "xMinYMax",
          "viewBox": "-65 160 100 100",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "mapBG",
              "width": NaN,
              "x": NaN,
              "xlinkHref": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "image",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <symbol
                  dangerouslySetInnerHTML={
                                    Object {
                                                      "__html": "Sup",
                                                    }
                  }
                  id="a"
                  preserveAspectRatio="xMidYMid"
                  viewBox="0 0 1 2"
/>,
              ],
              "className": "structureReference",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "a",
                "nodeType": "host",
                "props": Object {
                  "dangerouslySetInnerHTML": Object {
                    "__html": "Sup",
                  },
                  "id": "a",
                  "preserveAspectRatio": "xMidYMid",
                  "viewBox": "0 0 1 2",
                },
                "ref": null,
                "rendered": null,
                "type": "symbol",
              },
            ],
            "type": "defs",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "board": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "uniqueStructures": Array [
                Object {
                  "definition": Object {
                    "id": "",
                    "imageUrl": "",
                    "kind": "",
                    "length": 1,
                    "name": "",
                    "width": 1,
                  },
                  "definitionId": "",
                  "id": "a",
                  "image": Object {
                    "innerHTML": "Sup",
                  },
                  "imageHeight": 2,
                  "imageWidth": 1,
                  "name": "",
                  "xposition": 1,
                  "yposition": 1,
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "svg",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
